kolibri/plugins/coach/assets/src/views/lessons/LessonSummaryPage/sidePanels/LessonResourceSelection/subPages/PreviewSelectedResources/index.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+
+  <div>
+    <KCircularLoader v-if="loading && !contentNode" />
+    <PreviewContent
+      v-else
+      :currentContentNode="contentNode"
+      :ancestors="ancestors"
+      :isSelected="isSelected"
+      :questions="questions"
+      @addResource="handleAddResource"
+      @removeResource="handleRemoveResource"
+    />
+  </div>
+
+</template>
+
+
+<script>
+
+  import commonCoreStrings from 'kolibri/uiText/commonCoreStrings';
+  import useFetchContentNode from '../../../../../../../composables/useFetchContentNode';
+  import { coachStrings } from '../../../../../../common/commonCoachStrings';
+  import { PageNames } from '../../../../../../../constants/index';
+  import PreviewContent from './PreviewContent';
+
+  export default {
+    name: 'PreviewSelectedResources',
+    components: {
+      PreviewContent,
+    },
+    mixins: [commonCoreStrings],
+    setup(props) {
+      const { contentNode, ancestors, questions, loading } = useFetchContentNode(props.contentId);
+      const { manageLessonResourcesTitle$ } = coachStrings;
+
+      props.setTitle(manageLessonResourcesTitle$());
+      props.setGoBack(null);
+
+      return {
+        contentNode,
+        ancestors,
+        questions,
+        loading,
+      };
+    },
+    props: {
+      contentId: {
+        type: String,
+        required: true,
+      },
+      setTitle: {
+        type: Function,
+        default: () => {},
+      },
+      setGoBack: {
+        type: Function,
+        default: () => {},
+      },
+      selectedResources: {
+        type: Array,
+        required: true,
+      },
+    },
+    computed: {
+      isSelected() {
+        if (this.selectedResources && this.contentNode) {
+          return this.selectedResources.some(resource => resource.id === this.contentNode.id);
+        }
+        return false;
+      },
+      routeBack() {
+        const { params, query } = this.$route;
+        return {
+          name: PageNames.LESSON_SELECT_RESOURCES_TOPIC_TREE,
```

Comment:
We wont be always opening this view from the topic tree, we can open it from the bookmarks and from the search results. So it will be better to use a strategy like [this](https://github.com/allanoxdi/kolibri/blob/62ae0dbcff750fc36b3215b78e77de2dee8a4fa2/kolibri/plugins/coach/assets/src/views/common/reports/LearnerQuizPage.vue#L70) to store the prev route and navigate back to it here, and if there isnt a back route, we can navigate to `LESSON_RESOURCE_SELECTION_INDEX` instead, because we might not have a `topicId` query to render the topic tree view.